lib/nerves_hub_web/live/devices/show.ex

```diff
@@ -363,4 +364,13 @@ defmodule NervesHubWeb.Live.Devices.Show do
       device.connecting_code
     end
   end
+
+  defp extension_overrides(device, product) do
+    device.extensions
+    |> Map.from_struct()
+    |> Enum.filter(fn {extension, enabled} ->
+      enabled == false and Map.get(product.extensions, extension) == true
+    end)
+    |> Enum.map(&elem(&1, 0))
```

Comment:
I think we could save some cycles with this, thoughts?

```suggestion
    for {ext, enabled} <- device.extensions, !enabled and product.extensions[ext], into: [], do: ext
```

Reply:
Unfortunately we need to model extension settings being unset or we would have to choose a more aggressive default (always enable, always disable). So they can't just be true/false, they can also be nil. This is in line with Jon's original approach although maybe not intuitive.

So `and` will not work unless we do explicit comparisons to true/false which handles the nil cases for us.

Reply:
Ah, interesting okay. I updated my suggestion.